messageV2_test.go

```diff
@@ -0,0 +1,369 @@
+	stopMsg, ok := m.(*StreamStopMessageV2)
+	s.True(ok)
+
+	s.Equal(expected, stopMsg)
+}
+
+func TestStreamCommitV2Suite(t *testing.T) {
+	suite.Run(t, new(streamCommitSuite))
+}
+
+type streamCommitSuite struct {
+	messageSuite
+}
+
+func (s *streamCommitSuite) Test() {
+	msg := make([]byte, 1+4+1+8+8+8)
+	xid := s.newXid()
+	flags := uint8(0)
+	commitLSN := s.newLSN()
+	transactionEndLSN := s.newLSN()
+	commitTime, pgCommitTime := s.newTime()
+
+	msg[0] = 'c'
+	bigEndian.PutUint32(msg[1:], xid)
+	fmt.Printf("%+v\n", msg)
+	msg[5] = flags
+	bigEndian.PutUint64(msg[6:], uint64(commitLSN))
+	bigEndian.PutUint64(msg[14:], uint64(transactionEndLSN))
+	bigEndian.PutUint64(msg[22:], pgCommitTime)
+
+	expected := &StreamCommitMessageV2{
+		Xid:               xid,
+		Flags:             flags,
+		CommitLSN:         commitLSN,
+		TransactionEndLSN: transactionEndLSN,
+		CommitTime:        commitTime,
+	}
+	expected.msgType = MessageTypeStreamCommit
+
+	s.assertV1NotSupported(msg)
+
+	m, err := ParseV2(msg, false)
+	s.NoError(err)
+	streamCommitMsg, ok := m.(*StreamCommitMessageV2)
+	s.True(ok)
+	s.Equal(expected, streamCommitMsg)
+}
+
+func TestStreamAbortV2Suite(t *testing.T) {
+	suite.Run(t, new(streamAbortSuite))
+}
+
+type streamAbortSuite struct {
+	messageSuite
+}
+
+func (s *streamAbortSuite) Test() {
+	msg := make([]byte, 1+4+4)
+
+	xid := s.newXid()
+	subXid := s.newXid()
+
+	msg[0] = 'A'
+	bigEndian.PutUint32(msg[1:], xid)
+	bigEndian.PutUint32(msg[5:], subXid)
+
+	expected := &StreamAbortMessageV2{
+		Xid:    xid,
+		SubXid: subXid,
+	}
+	expected.msgType = MessageTypeStreamAbort
+
+	s.assertV1NotSupported(msg)
+
+	m, err := ParseV2(msg, false)
+	s.NoError(err)
+	streamAbortMsg, ok := m.(*StreamAbortMessageV2)
+	s.True(ok)
+	s.Equal(expected, streamAbortMsg)
+}
+
+func TestRelationMessageV2Suite(t *testing.T) {
+	suite.Run(t, new(relationMessageV2Suite))
+}
+
+type relationMessageV2Suite struct {
+	messageSuite
+}
+
+func (s *relationMessageV2Suite) Test() {
+	msg, expected := s.createRelationTestData()
+
+	msgV2, xid := s.insertXid(msg)
+
+	m, err := ParseV2(msgV2, true)
+	s.NoError(err)
+	relMsg, ok := m.(*RelationMessageV2)
+	s.True(ok)
+	s.Equal(xid, relMsg.Xid)
+	s.Equal(expected, &relMsg.RelationMessage)
+}
+
+func TestTypeMessageV2Suite(t *testing.T) {
+	suite.Run(t, new(typeMessageV2Suite))
+}
+
+type typeMessageV2Suite struct {
+	messageSuite
+}
+
+func (s *typeMessageV2Suite) Test() {
+	msg, expected := s.createTypeTestData()
+	msgV2, xid := s.insertXid(msg)
+
+	m, err := ParseV2(msgV2, true)
+	s.NoError(err)
+	typeMsg, ok := m.(*TypeMessageV2)
+	s.True(ok)
+	s.Equal(xid, typeMsg.Xid)
+	s.Equal(expected, &typeMsg.TypeMessage)
+}
+
+func TestInsertMessageV2Suite(t *testing.T) {
+	suite.Run(t, new(insertMessageV2Suite))
+}
+
+type insertMessageV2Suite struct {
+	messageSuite
+}
+
+func (s *insertMessageV2Suite) Test() {
+	msg, expected := s.createInsertTestData()
+	msgV2, xid := s.insertXid(msg)
+
+	m, err := ParseV2(msgV2, true)
```

Comment:
Add test case with `inStream = false` ?

Reply:
Good point, I'll add tests for `inStream = false` and update code mentioned in previous comments.
Thanks!